WearTilesKotlin/app/src/main/java/com/example/wear/tiles/messaging/MessagingTileRenderer.kt

```diff
@@ -51,223 +31,34 @@ class MessagingTileRenderer(context: Context) :
         state: MessagingTileState,
         deviceParameters: DeviceParametersBuilders.DeviceParameters
     ): LayoutElement {
-        return tileLayout(state, deviceParameters)
+        return messagingTileLayout(state, context, deviceParameters)
     }
 
+    /**
+     * If we want to display an image in our layout, we have to add the image resource to the
+     * [Resources.Builder].
+     */
     override fun Resources.Builder.produceRequestedResources(
         resourceResults: Map<Contact, Bitmap>,
         deviceParameters: DeviceParametersBuilders.DeviceParameters,
         resourceIds: MutableList<String>
     ) {
         if (resourceIds.isEmpty() || resourceIds.contains(ID_IC_SEARCH)) {
             addIdToImageMapping(
-                ID_IC_SEARCH,
-                imageResourceFrom(R.drawable.ic_search)
+                /* id = */ ID_IC_SEARCH,
+                /* image = */ drawableResToImageResource(R.drawable.ic_search_24)
             )
         }
 
-        // Add the scaled & cropped avatar images
         resourceResults.forEach { (contact, bitmap) ->
-            val imageResource = bitmapToImageResource(bitmap)
-            // Add each created image resource to the list
             addIdToImageMapping(
```

Comment:
From the chat, I think you are right, we should check these before returning.

Reply:
I was about to add them but it's checked already in MessagingTileService. I added a comment to make it clearer.